go/builder.go

```diff
@@ -33,7 +35,7 @@ func NewBuilder(initialSize int) *Builder {
 	b.head = UOffsetT(initialSize)
 	b.minalign = 1
 	b.vtables = make([]UOffsetT, 0, 16) // sensible default capacity
-
+	b.sharedStrings = make(map[string]UOffsetT)
```

Comment:
Only allocate the actual map if `CreateSharedString` gets called.. don't want to burder all the users with an extra allocation.

Reply:
Agreed.